### PR TITLE
feat(config-api, keycloak):  saml plugin to create trust client in DB and keycloak storage provider to jans store

### LINKED ISSUE #5148: feat(config-api): saml config-api plugin
New plugin to support SAML functionality.
Jans SAML to use Keycloak as identity Provider(IdP). New config-api plugin to expose endpoints to create and manage trust relationships and persist in DB.

Based on discussion with @uprightech 
1. config-saml-plugin is not using Keycloak API. 
2. config-saml-plugin will upload metadata file on server using existing `DocumentStore`
3. trust-relationship endpoint to expose methods to create and manage trust-relationship in DB
- GET: Need list of trust-relationship 
- POST: Create trust-relationship . 
- PUT: Update trust-relationship  
- DELETE: Delete trust-relationship  
4. Config to be declared in DB

### PR DESCRIPTION
**Pull request by pujavs**
### Prepare
 New config-api plugin to expose endpoints to create and manage trust relationships and persist in DB- issue#5148 
jans-keycloak-storage-api related changes
-------------------

#### Target issue
  
  
  
  
closes #5148 
          #6177 

#### Implementation Details
  

-------------------
### Test and Document the changes
- [ ] Static code analysis has been run locally and issues have been fixed
- [ ] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

### CODE REVIEW

File: docs/admin/install/setup.md [hunk 1]
```diff
@@ -86,9 +86,14 @@ To check usage of this script run help command
                      [-couchbase-admin-user COUCHBASE_ADMIN_USER] [-couchbase-admin-password COUCHBASE_ADMIN_PASSWORD]
                      [-couchbase-bucket-prefix COUCHBASE_BUCKET_PREFIX] [-couchbase-hostname COUCHBASE_HOSTNAME] [--no-data]
                      [--no-jsauth] [-ldap-admin-password LDAP_ADMIN_PASSWORD] [--no-config-api] [--no-scim] [--no-fido2]
+<<<<<<< HEAD
```
Comment (moabu) [inline docs/admin/install/setup.md]: watch out from merge operations. I can see this across your code.

File: docs/admin/install/setup.md [hunk 2]
```diff
@@ -86,9 +86,14 @@ To check usage of this script run help command
...
                      [--install-eleven] [--install-jans-link] [--with-casa] [--load-config-api-test]
                      [-config-patch-creds CONFIG_PATCH_CREDS] [-spanner-project SPANNER_PROJECT] [-spanner-instance SPANNER_INSTANCE]
                      [-spanner-database SPANNER_DATABASE]
+=======
+                     [--install-eleven] [--install-cache-refresh] [--install-jans-saml] [--load-config-api-test] [-config-patch-creds CONFIG_PATCH_CREDS]
+                     [-spanner-project SPANNER_PROJECT] [-spanner-instance SPANNER_INSTANCE] [-spanner-database SPANNER_DATABASE]
+>>>>>>> c49aae345dccea56c9b1af51f6d95ebc527201bf
```
Comment (moabu) [inline docs/admin/install/setup.md]: like here

File: docs/admin/install/setup.md [hunk 3]
```diff
@@ -163,8 +168,12 @@ Below are the optional arguments:
 | --no-scim | Do not install Scim Server |
 | --no-fido2 | Do not install Fido2 Server |
 | --install-eleven | Install Eleven Server |
+<<<<<<< HEAD
```
Comment (moabu) [inline docs/admin/install/setup.md]: and here

File: jans-config-api/plugins/keycloak-plugin/dependency-reduced-pom.xml [hunk 4]
```diff
@@ -0,0 +1,286 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <parent>
+    <artifactId>plugins</artifactId>
+    <groupId>io.jans.jans-config-api.plugins</groupId>
+    <version>1.0.17-SNAPSHOT</version>
```
Comment (moabu) [inline jans-config-api/plugins/keycloak-plugin/dependency-reduced-pom.xml]: use 1.0.19-SNAPSHOT

**Review by moabu — CHANGES_REQUESTED** [review]

File: docs/admin/install/setup.md (hunk 2 above)
Comment (pujavs) [reply docs/admin/install/setup.md]: Fixed.

File: docs/admin/install/setup.md (hunk 3 above)
Comment (pujavs) [reply docs/admin/install/setup.md]: Fixed.

File: jans-config-api/plugins/keycloak-plugin/dependency-reduced-pom.xml (hunk 4 above)
Comment (pujavs) [reply jans-config-api/plugins/keycloak-plugin/dependency-reduced-pom.xml]: Fixed.

**Review by moabu — DISMISSED** [review]

**Review by yuriyz — APPROVED** [review]

**Review by yurem — APPROVED** [review]

**Review by devrimyatar — DISMISSED** [review]

**Review by yuriyz — APPROVED** [review]

**Review by moabu — APPROVED** [review]

**Review by moabu — APPROVED** [review]

**Pull request merged**